tests/func-tests/monitoring_test.go

```diff
@@ -246,3 +275,139 @@ func getPrometheusUrl(cli kubecli.KubevirtClient) string {
+		Resource("nodes").Name(nodeName).
+		Suffix("proxy", "configz").
+		Do(context.TODO()).Raw()
+	if err != nil {
+		return 0, err
+	}
+
+	var kc Kubeletconfig
+	err = json.Unmarshal(resp, &kc)
+	if err != nil {
+		return 0, err
+	}
+
+	return kc.Kubeletconfig.NodeStatusMaxImages, nil
+}
+
+func verifyAlertExists(promClient promApiv1.API, alertName string) {
+	Eventually(func() *promApiv1.Alert {
+		alerts, err := promClient.Alerts(context.TODO())
+		Expect(err).ShouldNot(HaveOccurred())
+		alert := getAlertByName(alerts, alertName)
+		return alert
+	}, 60*time.Second, time.Second).ShouldNot(BeNil())
+}
+
+func createPod(virtCli kubecli.KubevirtClient, node *corev1.Node, imageName string) *corev1.Pod {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod",
+			Namespace: flags.KubeVirtInstallNamespace,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:    "test-container",
+					Image:   imageName,
+					Command: []string{"/bin/sh", "-c", "while [ ! -f /tmp/exit ]; do sleep 1; done"},
+				},
+			},
+			NodeName: node.Name,
+		},
+	}
+
+	_, err := virtCli.CoreV1().Pods(flags.KubeVirtInstallNamespace).Create(context.TODO(), pod, metav1.CreateOptions{})
+	Expect(err).ShouldNot(HaveOccurred())
+
+	return pod
+}
+
+func updatePodImage(virtCli kubecli.KubevirtClient, pod *corev1.Pod, imageName string) {
+	pod, err := virtCli.CoreV1().Pods(flags.KubeVirtInstallNamespace).Get(context.TODO(), pod.Name, metav1.GetOptions{})
+	Expect(err).ShouldNot(HaveOccurred())
+
+	pod.Spec.Containers[0].Image = imageName
+
+	_, err = virtCli.CoreV1().Pods(flags.KubeVirtInstallNamespace).Update(context.TODO(), pod, metav1.UpdateOptions{})
+	Expect(err).ShouldNot(HaveOccurred())
+}
+
+func waitForPodToBeReady(virtCli kubecli.KubevirtClient, pod *corev1.Pod) {
+	Eventually(func() bool {
+		pod, err := virtCli.CoreV1().Pods(flags.KubeVirtInstallNamespace).Get(context.TODO(), pod.Name, metav1.GetOptions{})
+		Expect(err).ShouldNot(HaveOccurred())
+		for _, cond := range pod.Status.Conditions {
+			if cond.Type == corev1.PodReady && cond.Status == corev1.ConditionTrue {
+				return true
+			}
+		}
+		return false
+	}, 60*time.Second, 1*time.Second).Should(BeTrue())
+}
+
+type DockerHubImageList struct {
+	baseImage    string
+	tagPrefix    string
+	currentIndex int
+	currentPage  int
+
+	result *struct {
+		Count   int    `json:"count"`
+		Next    string `json:"next"`
+		Results []struct {
+			Name string `json:"name"`
+		} `json:"results"`
+	}
+}
+
+func initDockerHubImageList(imageName string, tagPrefix string) *DockerHubImageList {
+	return &DockerHubImageList{
+		baseImage:    imageName,
+		tagPrefix:    tagPrefix,
+		currentIndex: 0,
+		currentPage:  0,
+		result:       nil,
+	}
+}
+
+func (dhil *DockerHubImageList) getNextImage() string {
+	if dhil.result == nil {
+		dhil.requestMoreImages(fmt.Sprintf("https://hub.docker.com/v2/repositories/library/%s/tags?name=%s", dhil.baseImage, dhil.tagPrefix))
```

Comment:
Can we please use quay.io instead of DockerHub?

Reply:
the problem is that in quay, I couldn't find any repository with small-sized images, but with a lot of versions. [busybox on quay](https://quay.io/repository/quay/busybox) only has the latest tag